cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -3122,15 +3122,37 @@
 (rule 5 (amode ty (iadd (sextend x @ (value_type $I32)) y) offset)
         (AMode.RegExtended (amode_add y offset) x (ExtendOp.SXTW)))
 
+;; `RegScaled*` rules where this matches an addition of an "index register" to a
+;; base register. The index register is shifted by the size of the type loaded
+;; in bytes to enable this mode matching.
+;;
+;; Note that this can additionally bundle an extending operation but the
+;; extension must happen before the shift. This will pattern-match the shift
+;; first and then if that succeeds afterwards try to find an extend.
```

Comment:
Can we add some tests for when the order of the shift vs extend is wrong in the filetests? IIRC we had some historical issues with amodes and doing the extend at the wrong time, so it would be good to get some future-proofing here.

Reply:
Indeed! I tried to make sure those existed [here](https://github.com/bytecodealliance/wasmtime/pull/6945/files#diff-24cb46c4d5e7fb0dd8b989bacc6ea06ea7ae25adf616836fd2564113a602a788R691-R710) and [here](https://github.com/bytecodealliance/wasmtime/pull/6945/files#diff-24cb46c4d5e7fb0dd8b989bacc6ea06ea7ae25adf616836fd2564113a602a788R594-R612)

Reply:
Thanks!